### ISSUE TITLE
root required to install and run indico / make it work in user level virtualenv

### ISSUE THREAD

**Issue by cederom** [indico/indico#2856]
Hey :-) Current INDICO architecture assumes user has the root privileges on the server (sudo in indico_initial_setup, Apache/Nginx/uWSGI configuration, etc). Most of the hosting companies, including cloud ones, usually provide only user level linux/unix shell, so no modification to Apache/Nginx/uWSGI or any other system level is permitted. INDICO seems to be interesting platform but its not possible to use it on a remote/cloud server with user level access.

Please make INDICO run on user level shell, using user provided virtualenv, with no web/wsgi server modifications required (i.e. use passenger).

**Comment by ThiefMaster** [comment]
This will also change with 2.0 (we'll strongly recommend using virtualenv for example).

However, you'll still need to be able to run a WSGI application. We don't intend to support workarounds such as running the application as CGI (would result in abysmal performance). UWSGI doesn't need root for example so you could run it as a non-root user.

FYI, you can already install indico without root privs if you tinker a bit with the script (remove the sudo call) and use a virtualenv.

**Comment by cederom** [comment]
Passenger WSGI would be nice to have working :-)

**Comment by ThiefMaster** [comment]
Any WSGI server should work (even with 1.x) - Indico doesn't (and won't) use features specific to a single WSGI container.

**Comment by cederom** [comment]
I did an virtualenv installation, modified the module, created configuration, my own passenger launcher, launched zdaemon, etc. The initial site showed up, but I got 500 on click. Application was not able to parse URLs I guess..

My provider use Passwnger WSGI that requires `./passenger_wsgi.py` and `./public/stuff/`. I have linked the `htdocs` to be `public` but something seems to be hardcoded in INDICO and changing `etc` to point the new location did not help..
